frontend/src/bundles/common/components/router-provider/router-provider.tsx

```diff
@@ -1,15 +1,38 @@
 import {
-    type RouteObject,
     createBrowserRouter,
     RouterProvider as LibraryRouterProvider,
 } from 'react-router-dom';
 
+import { ProtectedRoute } from '~/bundles/common/components/components.js';
+import { type RouteObject } from '~/bundles/common/types/types.js';
+
 type Properties = {
     routes: RouteObject[];
 };
+const RouterProvider: React.FC<Properties> = ({ routes }) => {
+    const mapRoutes = (routes: RouteObject[]): RouteObject[] => {
```

Comment:
```suggestion
  const mapRoutes = (
    routes: RouteObject[],
    isAlreadyProtected = false,
): LibraryRouteObject[] =>
    routes.map(({ isPrivate, element, children }): LibraryRouteObject => {
        const shouldProtect = !isAlreadyProtected && Boolean(isPrivate);

        return {
            ...(element && {
                element: shouldProtect ? (
                    <ProtectedRoute>{element}</ProtectedRoute>
                ) : (
                    element
                ),
            }),
            ...(children?.length && {
                children: mapRoutes(
                    children,
                    isAlreadyProtected || shouldProtect,
                ),
            }),
        };
    });
```
Currently already protected routes could be wrapped with ProtectedRoute component multiple times, maybe this will help.
But I'm not even sure if this works :sweat_smile:

Reply:
It actually fails to find even the root with this code))))
![image](https://github.com/BinaryStudioAcademy/bsa-winter-2023-2024-lime/assets/72942456/be62a037-0865-4dbd-9edd-d1bc6b9baeca)

Reply:
yeah :smile:  
path supposed to be moved from RouteObject to resulting object as well
But this still doesn't mean that this is works)

Reply:
Did as you said, now have to work fine)